src/User.php

```diff
@@ -6468,4 +6577,44 @@ public static function getUserByForgottenPasswordToken(string $token): ?User
 
         return $user;
     }
+
+    /**
+     * Create a new user from an email address
+     *
+     * @param email The email address of the user.
+     *
+     * @return The user object.
+     */
+    public static function createUserFromMail($email)
+    {
+        global $DB;
+
+        $iterator = $DB->request([
+            'SELECT' => 'id',
+            'FROM'   => UserEmail::getTable(),
+            'WHERE'  => [
+                'email' => $email
+            ]
+        ]);
+
+        if (count($iterator) > 0) {
+            return false;
+        }
+
+        $user = new self();
+        $user->add([
+            'name'           => $email,
+            'realname'       => $email,
+            'password'       => Toolbox::getRandomString(16),
```

Comment:
You should probably move the randomization of the password in the `prepareInputForAdd` method. This would permit to use this feature from API.